core/trino-main/src/main/java/io/trino/operator/OperatorFactories.java

```diff
@@ -56,32 +61,62 @@ OperatorFactory probeOuterJoin(
 
-    OperatorFactory lookupOuterJoin(
-            int operatorId,
-            PlanNodeId planNodeId,
-            JoinBridgeManager<?> lookupSourceFactory,
-            boolean waitForBuild,
-            boolean hasFilter,
-            boolean spillingEnabled,
-            List<Type> probeTypes,
-            List<Integer> probeJoinChannel,
-            OptionalInt probeHashChannel,
-            Optional<List<Integer>> probeOutputChannels,
-            OptionalInt totalOperatorsCount,
-            PartitioningSpillerFactory partitioningSpillerFactory,
-            BlockTypeOperators blockTypeOperators);
+    public static class JoinOperatorType
+    {
+        private final JoinNode.Type type;
+        private final boolean outputSingleMatch;
+        private final boolean waitForBuild;
 
-    OperatorFactory fullOuterJoin(
-            int operatorId,
-            PlanNodeId planNodeId,
-            JoinBridgeManager<?> lookupSourceFactory,
-            boolean hasFilter,
-            boolean spillingEnabled,
-            List<Type> probeTypes,
-            List<Integer> probeJoinChannel,
-            OptionalInt probeHashChannel,
-            Optional<List<Integer>> probeOutputChannels,
-            OptionalInt totalOperatorsCount,
-            PartitioningSpillerFactory partitioningSpillerFactory,
-            BlockTypeOperators blockTypeOperators);
+        public static JoinOperatorType innerJoin(boolean outputSingleMatch, boolean waitForBuild)
+        {
+            return new JoinOperatorType(JoinNode.Type.INNER, outputSingleMatch, waitForBuild);
+        }
+
+        public static JoinOperatorType probeOuterJoin(boolean outputSingleMatch)
+        {
+            return new JoinOperatorType(JoinNode.Type.LEFT, outputSingleMatch, false);
+        }
+
+        public static JoinOperatorType lookupOuterJoin(boolean waitForBuild)
+        {
+            return new JoinOperatorType(JoinNode.Type.RIGHT, false, waitForBuild);
+        }
+
+        public static JoinOperatorType fullOuterJoin()
+        {
+            return new JoinOperatorType(JoinNode.Type.FULL, false, false);
+        }
+
+        private JoinOperatorType(JoinNode.Type type, boolean outputSingleMatch, boolean waitForBuild)
+        {
+            this.type = requireNonNull(type, "type is null");
+            this.outputSingleMatch = outputSingleMatch;
+            this.waitForBuild = waitForBuild;
+        }
+
+        public JoinNode.Type getType()
+        {
+            return type;
+        }
+
+        public boolean isOutputSingleMatch()
+        {
+            return outputSingleMatch;
+        }
+
+        public boolean isWaitForBuild()
+        {
+            return waitForBuild;
+        }
+
+        public JoinType toJoinType()
+        {
+            return switch (getType()) {
+                case INNER -> INNER;
+                case LEFT -> PROBE_OUTER;
+                case RIGHT -> LOOKUP_OUTER;
+                case FULL -> FULL_OUTER;
```

Comment:
Does this syntax prevent not matched case?

Reply:
yes